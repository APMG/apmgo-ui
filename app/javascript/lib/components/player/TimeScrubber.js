// @flow
import React from 'react'
import { dispatch, connect } from 'react-redux'
import { timeScrubberChange, updatePlayTime } from '../../redux/audio-player'
import Slider from 'rc-slider'

type TimeScrubberProps = {
  paused: boolean,
  currentTime: number,
  duration: number,
  updatePlayTime: (playtime: number) => {},
  timeScrubberChange: (currentTime: number) => {}
}

export default class TimeScrubberPresenter extends React.Component {

  props: TimeScrubberProps
  state: {
    timeDraggingPoint: number|false
  }

  constructor(props: TimeScrubberProps) {
    super(props)
    this.state = {
      timeDraggingPoint: false
    }
  }

  timeScrubberValue() {
    if(!this.props.duration || isNaN(this.props.duration)) {
      return 0
    }

    let val = this.props.currentTime // / this.props.duration
    return val // <= 1 ? val : 1
  }

  // EVENT HANDLERS

  _timeChange(newValue: number) {
    if (this.state.timeDraggingPoint) {
      this.setState({
        timeDraggingPoint: newValue
      })
      this.props.updatePlayTime(newValue) //this.props.duration * )
    } else {
      this.setState({
        timeDraggingPoint: newValue
      })
    }
  }

  _timeChangeDragStop() {
    if (!this.state.timeDraggingPoint) {
      return
    }

    if (this.state.timeDraggingPoint) {
      let currentTime = this.state.timeDraggingPoint //* this.props.duration
      this.props.timeScrubberChange(currentTime)
    }

    this.setState({
      timeDraggingPoint: false
    })
  }

  render() {
    return (
      <div>
        <Slider
          max={this.props.duration}
          min={0}
          value={this.timeScrubberValue()}
          onChange={ this._timeChange.bind(this) }
          onAfterChange={this._timeChangeDragStop.bind(this) }
        />
      </div>
    )
  }
}
