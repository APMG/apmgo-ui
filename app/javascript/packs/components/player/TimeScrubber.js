// @flow
import React from 'react'
import { dispatch, connect } from 'react-redux'
import { timeScrubberChange, updatePlayTime } from '../../redux/audio-player'
import Slider from 'material-ui/Slider'

type TimeScrubberProps = {
  paused: boolean,
  currentTime: number,
  duration: number,
  updatePlayTime: () => {},
  timeScrubberChange: () => {}
}

export class TimeScrubberPresenter extends React.Component {

  props: TimeScrubberProps
  state: {
    timeDraggingPoint: number|false
  }

  constructor(props: TimeScrubberProps) {
    super(props)
    this.state = {
      timeDraggingPoint: false
    }
  }

  timeScrubberValue() {
    if(!this.props.duration || isNaN(this.props.duration)) {
      return 0
    }

    let val = this.props.currentTime / this.props.duration
    return val <= 1 ? val : 1
  }

  // EVENT HANDLERS

  _timeChange(event: Event, newValue: number) {
    if (this.state.timeDraggingPoint) {
      this.setState({
        timeDraggingPoint: newValue
      })
      this.props.updatePlayTime(this.props.duration * newValue)
    } else {
      this.setState({
        timeDraggingPoint: newValue
      })
    }
  }

  _timeChangeDragStop() {
    if (!this.state.timeDraggingPoint) {
      return
    }

    if (this.state.timeDraggingPoint) {
      let currentTime = this.props.duration * this.state.timeDraggingPoint
      this.props.timeScrubberChange(currentTime)
    }

    this.setState({
      timeDraggingPoint: false
    })
  }

  render() {
    return (
      <div>
        <Slider
          value={ this.timeScrubberValue() }
          onChange={ this._timeChange.bind(this) }
          onDragStop={this._timeChangeDragStop.bind(this) }
        />
      </div>
    )
  }
}

// const mapDispatchToProps = (dispatch, ownProps) => {
//   return {
//     updatePlayTime: (currentTime) => {
//       dispatch(updatePlayTime(ownProps.item_id, Math.ceil(currentTime)))
//     },
//     timeScrubberChange: (currentTime) => {
//       dispatch(timeScrubberChange(ownProps.item_id, Math.ceil(currentTime)))
//     }
//   }
// }

// const mapStateToProps = (state) => {
//   return {
//     currentTime: state.audioPlayer.currentTime,
//     duration: state.audioPlayer.duration,
//     paused: state.audioPlayer.paused
//   }
// }

export default TimeScrubberPresenter //connect(mapStateToProps, mapDispatchToProps)(TimeScrubberPresenter)
