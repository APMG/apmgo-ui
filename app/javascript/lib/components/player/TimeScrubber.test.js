
import React from 'react'
import { getSnapshotJson } from '../../__tests__/testHelpers'
import TimeScrubber from './TimeScrubber'

describe('Time Scrubber Component', () => {
  describe('Presenter', () => {
    it('Renders', () => {
      let tree = getSnapshotJson(<TimeScrubber paused />)
      expect(tree).toMatchSnapshot()
    })
  })
})
