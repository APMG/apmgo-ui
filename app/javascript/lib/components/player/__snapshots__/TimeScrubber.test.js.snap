// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Time Scrubber Component Presenter Renders 1`] = `
<div>
  <input
    aria-label="Time Scrubber"
    className="AudioPlayer__scrubber___2VRHx"
    max={undefined}
    min={0}
    onInput={[Function]}
    onMouseUp={[Function]}
    type="range"
    value={0}
  />
</div>
`;
